packages/medusa/src/services/cart.ts

```diff
@@ -229,10 +229,9 @@ class CartService extends TransactionBaseService<CartService> {
           totals.discount_total = this.totalsService_.getDiscountTotal(cart)
           break
         case "tax_total":
-          totals.tax_total = await this.totalsService_.getTaxTotal(
-            cart,
-            options.force_taxes
-          )
+          totals.tax_total = await this.totalsService_
+            .withTransaction(this.manager_)
```

Comment:
todo: thinking that we should remove this - there is no guarantee that the manager is a transaction manager and since we are only reading this it shouldn't be a big concern to have this in a transaction. 

What was the idea here?

Reply:
I dont think that we should remove it, when we are updating a cart, an order or something, we are in a transaction, which means that normally if we call this without transaction then it could compute on old values since the current transaction is not committed. This is even more true now that we wrap the action in a transaction from the handler.Also, passing the manager ensure that it either use the injected manager and therefore run the query as usual out of a transaction, or the manager will  be a transactional one when we are in a transaction context and therefore use the updated values. Also the manager hold a value is_transaction or something similar so there shouldn’t be a problem

Wdyt?

Reply:
In the end, i think that every service should be called withTransaction and either it will be used with the default manager outside of a transaction or with a transactionManager. Because you never know it the called method will have to fetch the data that have been updated in a current running transaction. Wdyt?